Source/Editor/Content/Create/PrefabCreateEntry.cs

```diff
@@ -117,7 +118,8 @@ public enum WidgetMode
 
             private static bool IsValid(Type type)
             {
-                return (type.IsPublic || type.IsNestedPublic) && !type.IsAbstract && !type.IsGenericType;
+                var controlTypes = Editor.Instance.CodeEditing.Controls.Get();
+                return (type.IsPublic || type.IsNestedPublic) && !type.IsAbstract && !type.IsGenericType && controlTypes.Any(c => c.Type == type);
```

Comment:
UI is C#-only so we can ignore Script Type and check for C# controls quicker via:
```cs
controlTypes.Contains(new ScriptType(type))
```